nmrcraft/evaluation/evaluation.py

```diff
@@ -46,3 +60,60 @@ def model_evaluation(
         fpr,
         tpr,
     )
+
+
+def model_evaluation_nD(
+    model: BaseEstimator,
```

Comment:
Very nice. Also a note for final review (I’m creating an issue out of this: add type hints for all functions and classes. The purpose of this is to enforce that a function takes inputs only of a certain type, and also only outputs a certain type.

instead of 

```
def confusion_matric_plotter_or_whatever(target, full=True):
   return 1+1
```

Write

```
def confusion_matric_plotter_or_whatever(target: np.array, full: bool = True) -> int:
   return 1+1
   ```
   
   This seems a bit useless at first but its a game changer when writing code that takes in many different inputs, and saves some amount of error messages!